### PR TITLE
Multiple LFS improvements

### LINKED ISSUE #10665: git lfs pushed files are empty
- Gitea version (or commit ref): 1.11.2
- Git version: 2.17.1; git-lfs 2.3.4
- Operating system: Ubuntu 18.04
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant; LFS not enabled on try.gitea.io !
- Log gist: https://gist.github.com/markus-seidl/c9f51da5e5855d811581970517e0b339

## Description

I try to push files from a client repository that has LFS enabled. LFS is enabled on the server and the push succeeds. Also the settings page of the repository lists the files as stored in the repository with their correct sizes.
However, on the next 
* clone
* Zip download of the repository
* Or direct download via web ui
the files are empty or result in a 500 error. The 500 is, because the files are not there.

### PR DESCRIPTION
**Pull request by zeripath**
* Radically increase the logging within the LFS server and prevent the passing of internal error information to clients
* Handle cursoring in LFS Lock querying.
* Use a standardised jwt.Claims object rather than passing generic map

Close #10665

### CODE REVIEW

File: modules/lfs/server.go [hunk 1]
```diff
@@ -91,6 +99,7 @@ func isOidValid(oid string) bool {
 // ObjectOidHandler is the main request routing entry point into LFS server functions
 func ObjectOidHandler(ctx *context.Context) {
 	if !setting.LFS.StartServer {
+		log.Debug("Attempt to access LFS server but LFS server is disabled")
```
Comment (guillep2k) [inline modules/lfs/server.go:102]:
```suggestion
		log.Info("Attempt to access LFS server but LFS server is disabled")
```

File: modules/lfs/locks.go [hunk 2]
```diff
@@ -19,10 +19,12 @@ import (
 //checkIsValidRequest check if it a valid request in case of bad request it write the response to ctx.
 func checkIsValidRequest(ctx *context.Context) bool {
 	if !setting.LFS.StartServer {
+		log.Debug("Attempt to access LFS server but LFS server is disabled")
```
Comment (guillep2k) [inline modules/lfs/locks.go:22]:
```suggestion
		log.Info("Attempt to access LFS server but LFS server is disabled")
```

File: modules/lfs/server.go [hunk 3]
```diff
@@ -255,18 +279,22 @@ func PostHandler(ctx *context.Context) {
...
+	if err := enc.Encode(Represent(rv, meta, meta.Existing, true)); err != nil {
+		log.Error("Failed to encode representation as json. Error: %v", err)
+	}
 	logRequest(ctx.Req, sentStatus)
 }
 
 // BatchHandler provides the batch api
 func BatchHandler(ctx *context.Context) {
 	if !setting.LFS.StartServer {
+		log.Debug("Attempt to access LFS server but LFS server is disabled")
```
Comment (guillep2k) [inline modules/lfs/server.go:291]:
```suggestion
		log.Info("Attempt to access LFS server but LFS server is disabled")
```

File: modules/lfs/server.go [hunk 4]
```diff
@@ -205,11 +225,13 @@ func getMetaHandler(ctx *context.Context) {
 // PostHandler instructs the client how to upload data
 func PostHandler(ctx *context.Context) {
 	if !setting.LFS.StartServer {
+		log.Debug("Attempt to access LFS server but LFS server is disabled")
```
Comment (guillep2k) [inline modules/lfs/server.go:228]:
```suggestion
		log.Info("Attempt to access LFS server but LFS server is disabled")
```

File: modules/lfs/server.go [hunk 5]
```diff
@@ -356,11 +394,13 @@ func PutHandler(ctx *context.Context) {
 // VerifyHandler verify oid and its size from the content store
 func VerifyHandler(ctx *context.Context) {
 	if !setting.LFS.StartServer {
+		log.Debug("Attempt to access LFS server but LFS server is disabled")
```
Comment (guillep2k) [inline modules/lfs/server.go:397]:
```suggestion
		log.Info("Attempt to access LFS server but LFS server is disabled")
```

File: cmd/serv.go [hunk 6]
```diff
@@ -213,12 +214,14 @@ func runServ(c *cli.Context) error {
...
-		claims := jwt.MapClaims{
-			"repo": results.RepoID,
-			"op":   lfsVerb,
-			"exp":  now.Add(setting.LFS.HTTPAuthExpiry).Unix(),
-			"nbf":  now.Unix(),
-			"user": results.UserID,
+		claims := lfs.Claims{
+			StandardClaims: jwt.StandardClaims{
+				ExpiresAt: now.Add(setting.LFS.HTTPAuthExpiry).Unix(),
+				NotBefore: now.Unix(),
```
[the remark this answers is not in the document]
Comment (zeripath) [reply cmd/serv.go:220]: So this value is created on the server and checked on the server. If we had a cluster of servers - then skew would be more important but the skew would have to be big enough to account for the JWT claim being passed back to the client and the client reopening a http request to the new server. 

Not sure. If there's that much skew between servers we're probably into other things failing?

File: modules/lfs/server.go (hunk 1 above)
Comment (zeripath) [reply modules/lfs/server.go:102]: I dunno I think Info is too high - there's no error here except on the user's behalf and the router logs will clearly indicate a 404 for the lfs pages.

**Review by guillep2k — APPROVED** [review]
👍

**Review by lafriks — APPROVED** [review]

**Pull request merged**